File: Javascript/Algorithms/Sorting/selectionSort.js
function selectionSort(arr) {
  const swap = (arr, index1, index2) =>
    ([arr[index1], arr[index2]] = [arr[index2], arr[index1]]);

  for (let i = 0; i < arr.length; i++) {
    let lowest = i;
    for (let j = i + 1; j < arr.length; j++) {
      if (arr[lowest] > arr[j]) {
        lowest = j;
      }
    }
    if (i !== lowest) swap(arr, i, lowest);
  }

  return arr;
}

function selectionSort2(array) { // selection sort to stop iteration if array is already sorted.
  for (let i = 0; i < array.length; i++) {
    let minValueIndex = i;
    let alreadySorted = 1;
    for (let j = i + 1; j < array.length; j++) {
      if (array[j] < array[i])
        minValueIndex = j;
      if (array[j] < array[j - 1])
        alreadySorted = 0;
    }
    if (alreadySorted)
      break;
    if (minValueIndex !== i)
      [array[i], array[minValueIndex]] = [array[minValueIndex], array[i]];
  }
  return array;
}


const myArray = [0,4,22,45,12,15,17]
selectionSort(myArray);
selectionSort2(myArray)

File: C++/Algorithms/Searching/selection_sort.cpp
#include<iostream>
using namespace std;

void selection_sort(int a[],int n)
{
    for(int i=0;i<n;i++)
    {
        int min =a[i];
        int minIndex=i;
        for(int j=i+1;j<n;j++)
        {
            if(a[j]<min)
            {
                min=a[j];
                minIndex = j;
            }
        } 

        swap(a[minIndex],a[i]);  
    }
}


int main()
{
    int a[6] = {1,4,5,3,12,-3};
    int  n=6;

    selection_sort(a,n);
    cout<<"Sorted Array is : "<<endl;

    for(int i=0;i<6;i++)
    {

        cout<<a[i]<<" ";
    }


    return 0;
}